core/drivers/imx_i2c.c

```diff
@@ -132,10 +149,17 @@ static void i2c_set_prescaler(uint8_t bid, uint32_t bps)
 
 static void i2c_set_bus_speed(uint8_t bid, int bps)
 {
-	/* Enable the clock */
-	io_write32(i2c_clk.base.va + CCM_CCGRx_SET(i2c_clk.i2c[bid]),
-		   CCM_CCGRx_ALWAYS_ON(0));
-
+	vaddr_t addr = i2c_clk.base.va;
+	uint32_t val = 0;
+
+#if defined(CFG_MX8MM)
+		addr += CCM_CCGRx_SET(i2c_clk.i2c[bid]);
+		val = CCM_CCGRx_ALWAYS_ON(0);
+#elif defined(CFG_MX6ULL)
+		addr += i2c_clk.i2c[bid];
+		val = i2c_clk.cgrbm[bid] | io_read32(addr);
+#endif
```

Comment:
I think an `#else` would be welcome here, likely to emit a build error.

Reply:
yes, lets do that

Reply:
`#else` should be part of commit "drivers: imx_i2c: add support for imx6ull",
not of commit "drivers: imx_i2c: add support for imx6ull".

not in commit "drivers: imx_i2c: move utility macros".